app/src/main/java/com/github/alfabravo2013/readyforexams/presentation/home/HomeViewModel.kt

```diff
@@ -1,6 +1,44 @@
 package com.github.alfabravo2013.readyforexams.presentation.home
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.alfabravo2013.readyforexams.domain.home.LoadChecklistUseCase
+import com.github.alfabravo2013.readyforexams.domain.models.Checklist
+import com.github.alfabravo2013.readyforexams.util.SingleLiveEvent
+import kotlinx.coroutines.launch
 
-class HomeViewModel : ViewModel() {
-}
\ No newline at end of file
+class HomeViewModel(
+    private val loadChecklistUseCase: LoadChecklistUseCase
+) : ViewModel() {
+    private val _checklists = MutableLiveData<List<Checklist>>().apply { emptyList<Checklist>() }
+    val checklists: LiveData<List<Checklist>> get() = _checklists
```

Comment:
No need for this checkLists events. You already have onEvent. You can create a data class inside there call ShowCheckList(val checkList: List<Checklist>) : OnEvent.